src/screens/LunchScreen.js

```diff
@@ -0,0 +1,155 @@
+  FlatList,
+  RefreshControl,
+  StyleSheet,
+  TouchableOpacity
+} from 'react-native';
+
+import { colors, consts, normalize, texts } from '../config';
+import {
+  BoldText,
+  Button,
+  ConnectedImagesCarousel,
+  HeaderLeft,
+  Icon,
+  LunchSection,
+  RegularText,
+  SafeAreaViewFlex,
+  Wrapper,
+  WrapperRow,
+  WrapperWithOrientation
+} from '../components';
+import { arrowLeft, arrowRight } from '../icons';
+import { useMatomoTrackScreenView } from '../hooks';
+import { graphqlFetchPolicy } from '../helpers';
+import moment from 'moment';
+import { NetworkContext } from '../NetworkProvider';
+import { useQuery } from 'react-apollo';
+import { getQuery, QUERY_TYPES } from '../queries';
+
+const { MATOMO_TRACKING } = consts;
+
+export const LunchScreen = ({ navigation }) => {
+  const [poiId, setPoiId] = useState(navigation.getParam('poiId'));
+
+  const [date, setDate] = useState(moment());
+
+  const { isConnected, isMainserverUp } = useContext(NetworkContext);
+  const fetchPolicy = graphqlFetchPolicy({ isConnected, isMainserverUp });
+
+  const currentDate = moment(date).format('YYYY-MM-DD');
+
+  const variables = {
+    dateRange: [currentDate, currentDate]
+  };
+
+  const { data, loading, refetch } = useQuery(getQuery(QUERY_TYPES.LUNCHES), {
+    fetchPolicy,
+    variables
+  });
+
+  useMatomoTrackScreenView(MATOMO_TRACKING.SCREEN_VIEW.LUNCH);
+
+  const onPressNext = useCallback(() => {
+    setDate((oldDate) => moment(oldDate).add(1, 'day'));
+  }, [setDate]);
+
+  const onPressPrevious = useCallback(() => {
+    setDate((oldDate) => moment(oldDate).subtract(1, 'day'));
+  }, [setDate]);
+
+  const renderItem = useCallback(
+    ({ item }) => <LunchSection lunchOfferData={item} navigation={navigation} />,
+    [navigation]
+  );
+
+  let lunchData = data?.[QUERY_TYPES.LUNCHES];
+
+  if (poiId) {
+    lunchData?.filter((item) => item?.pointOfInterest?.id === poiId);
+  }
+
+  const ListHeaderComponent = (
+    <>
+      <ConnectedImagesCarousel
+        navigation={navigation}
+        publicJsonFile="lunchCarousel"
+        refreshTimeKey="publicJsonFile-lunchCarousel"
+      />
+      <WrapperWithOrientation>
+        <Wrapper>
+          <WrapperRow>
+            <TouchableOpacity onPress={onPressPrevious} style={styles.left}>
+              <Icon xml={arrowLeft(colors.primary)} />
+            </TouchableOpacity>
+            <BoldText big>{date.format('DD.MM.YYYY')}</BoldText>
+            <TouchableOpacity onPress={onPressNext} style={styles.right}>
+              <Icon xml={arrowRight(colors.primary)} />
+            </TouchableOpacity>
+          </WrapperRow>
+        </Wrapper>
+      </WrapperWithOrientation>
+    </>
+  );
+
+  const ListEmptyComponent = (
+    <Wrapper>
+      <RegularText>{texts.lunch.noOffers}</RegularText>
+    </Wrapper>
+  );
+
+  const ListFooterComponent = (
+    <Wrapper>
+      <Button title={texts.lunch.showAll} onPress={() => setPoiId(undefined)} />
+    </Wrapper>
```

Comment:
button works better here

![Bildschirmfoto 2021-02-09 um 18 08 16](https://user-images.githubusercontent.com/1942953/107400190-dd7c9200-6b01-11eb-90e7-1279c3e26b22.png)

Reply:
what should happen in the app on click if there are more lunch offers for more points of interest?

Reply:
This button will show all lunch offers for that day.
you only get this button of you navigate to it through a poi that has lunches.
otherwise we will already load all lunches for a given date at once.

we could add pagination, but i do not know if it will be necessary. that depends on how many lunch offers there will be

Reply:
ok, then we will wait for the data to become too much